src/lib.rs

```diff
@@ -176,15 +176,19 @@ impl_arbitrary_for_floats! {
 impl Arbitrary for char {
     fn arbitrary(u: &mut Unstructured<'_>) -> Result<Self> {
         use std::char;
-        const CHAR_END: u32 = 0x0011_000;
-        // The size of the surrogate blocks
-        const SURROGATES_START: u32 = 0xD800;
-        let mut c = <u32 as Arbitrary>::arbitrary(u)? % CHAR_END;
+
+        // The largest u32 that is a valid unicode character.
+        const CHAR_END: u32 = 0x0011_000 - 1;
+
+        // The size of the surrogate blocks.
+        const SURROGATES_SIZE: u32 = 0xD800;
+
+        let c: u32 = u.int_in_range(0..=CHAR_END)?;
```

Comment:
I'm not sure if we should do this, this seems slower.

Reply:
Also, the older method slurps up chars as utf32, this will slurp them up as basically utf24 (not a thing that actually exists). The high byte in UTF32 is always null. This seems weird. If we want chars to parse as variable length we should attempt to parse them as utf8 instead.

Reply:
Sounds good, will revert this commit.